modules/simul/src/main/SimulApi.scala

```diff
@@ -74,9 +75,13 @@ final class SimulApi(
       position = setup.realPosition,
       color = setup.color.some,
       text = setup.text,
+      estimatedStartAt = setup.estimatedStartAt,
       team = setup.team,
       featurable = some(~setup.featured && me.canBeFeatured)
     )
+    if (prev.estimatedStartAt.isDefined && !simul.estimatedStartAt.isDefined) {
+      repo.removeEstimatedStartAt(simul)
+    }
```

Comment:
the `$unset` should be part of `SimulRepo.update` instead

Reply:
Ah, that sounds good, not adding new methods.

In my update of this code I have a couple of doubts,
1. I worry a bit what would happen if there would be some error while updating the database.
Could this "irrelevant" optional field cause any troubles with the "real" update, I wonder...
2. The unset will now  be invoked everytime there's no estimatedStartAt defined in the simul.
Updates "should" not happen often, but maybe it is silly to try to update when there is no field.

98240be

Reply:
Ok,
so looking a bit more at this - I can make it look a little bit better with the following -

    def update(simul: Simul) =
      coll.update
        .one(
          $id(simul.id),
          if (! simul.estimatedStartAt.isDefined) {
            $unset("estimatedStartAt") ++ $set(SimulBSONHandler writeTry simul get)
          } else {
            $set(SimulBSONHandler writeTry simul get)
          }
        )
        .void

But this could probably be more simpliefied - feels bad to repeat the same "$set" code twice...

Reply:
Replaced the if-else with the ??-Lilaism to extract the Option

a3a33ba